bentoml/deployment/sagemaker.py

```diff
@@ -0,0 +1,181 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import shutil
+import base64
+import subprocess
+import logging
+import re
+from six.moves.urllib.parse import urlparse
+
+import boto3
+import docker
+
+from bentoml.archive import load
+from bentoml.deployment.utils import generate_bentoml_deployment_snapshot_path
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_REGION = 'us-west-2'
+DEFAULT_INSTANCE_TYPE = 'ml.m4.xlarge'
+DEFAULT_INITIAL_INSTANCE_COUNT = 1
+
+
+def strip_scheme(url):
+    """ Stripe url's schema
+    e.g.   http://some.url/path -> some.url/path
+    :param url: String
+    :return: String
+    """
+    parsed = urlparse(url)
+    scheme = "%s://" % parsed.scheme
+    return parsed.geturl().replace(scheme, '', 1)
+
+
+def generate_aws_compatible_string(item):
+    pattern = re.compile('[^a-zA-Z0-9-]|_')
+    return re.sub(pattern, '-', item)
+
+
+def get_arn_role_from_current_user():
+    sts_client = boto3.client("sts")
+    identity = sts_client.get_caller_identity()
+    sts_arn = identity["Arn"]
+    sts_arn_list = sts_arn.split(':')
+    type_role = sts_arn_list[-1].split('/')
+    iam_client = boto3.client("iam")
+    if type_role[0] == 'user':
+        role_list = iam_client.list_roles()
+        arn = None
+        for role in role_list['Roles']:
+            policy_document = role['AssumeRolePolicyDocument']
+            statement = policy_document['Statement'][0]
+            if statement['Effect'] == 'Allow' and statement['Principal'][
+                    'Service'] == 'sagemaker.amazonaws.com':
+                arn = role['Arn']
+        if arn is None:
+            raise ValueError(
+                "Can't find proper Arn role for Sagemaker, please create one and try again")
+        return arn
+    elif type_role[0] == 'role':
+        role_response = iam_client.get_role(RoleName=type_role[1])
+        return role_response["Role"]["Arn"]
+
+
+def create_push_image_to_ecr(bento_service, snapshot_path):
+    """Create BentoService sagemaker image and push to AWS ECR
+
+    :param bento_service: Bento Service
+    :param snapshot_path: Path
+    :return: ecr_tag: String
+    """
+
+    # Example:
+    # https://github.com/awslabs/amazon-sagemaker-examples/blob/master/advanced_functionality/scikit_bring_your_own/container/build_and_push.sh
+    # 1. get aws account info and login ecr
+    # 2. create ecr repository, if not exist
+    # 3. build tag and push docker image
+
+    ecr_client = boto3.client('ecr')
+    token = ecr_client.get_authorization_token()
+    logger.info('Getting docker login info from AWS')
+    username, password = base64.b64decode(
+        token['authorizationData'][0]['authorizationToken']).decode('utf-8').split(":")
+    registry_url = token['authorizationData'][0]['proxyEndpoint']
+
+    # https://github.com/docker/docker-py/issues/2256
+    subprocess.call(['docker', 'login', '-u', username, '-p', password, registry_url])
```

Comment:
make sure to check if "docker" command is available similar to what we did with serverless command

Reply:
would be great to change this back to docker_client call once they fixed that issue. that way we don't need to rely on docker client cli installed on user machine.

Reply:
done